src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_ecc_import_export.c

```diff
@@ -0,0 +1,547 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_ecc_import_export.h"
+#include "pal_utilities.h"
+#include "pal_jni.h"
+#include "pal_eckey.h"
+#include "pal_bignum.h"
+
+int32_t CryptoNative_GetECKeyParameters(
+    const EC_KEY* key,
+    int32_t includePrivate,
+    jobject* qx, int32_t* cbQx,
+    jobject* qy, int32_t* cbQy,
+    jobject* d, int32_t* cbD)
+{
+    assert(qx != NULL);
+    assert(cbQx != NULL);
+    assert(qy != NULL);
+    assert(cbQy != NULL);
+    assert(d != NULL);
+    assert(cbD != NULL);
+
+    JNIEnv* env = GetJNIEnv();
+
+    // Get the public key
+    jobject publicKey = (*env)->CallObjectMethod(env, key->keyPair, g_keyPairGetPublicMethod);
+
+    jobject Q = (*env)->CallObjectMethod(env, publicKey, g_ECPublicKeyGetW);
+
+    (*env)->DeleteLocalRef(env, publicKey);
+
+    jobject xBn = (*env)->CallObjectMethod(env, Q, g_ECPointGetAffineX);
+    jobject yBn = (*env)->CallObjectMethod(env, Q, g_ECPointGetAffineY);
+    
+    (*env)->DeleteLocalRef(env, Q);
+
+    // Success; assign variables
+    *qx = ToGRef(env, xBn); *cbQx = CryptoNative_GetBigNumBytes(xBn); xBn = NULL;
+    *qy = ToGRef(env, yBn); *cbQy = CryptoNative_GetBigNumBytes(yBn); yBn = NULL;
+    if (*cbQx == FAIL || *cbQy == FAIL) goto error;
+
+    if (includePrivate)
+    {
+        jobject privateKey = (*env)->CallObjectMethod(env, key->keyPair, g_keyPairGetPrivateMethod);
+
+        jobject dBn = (*env)->CallObjectMethod(env, privateKey, g_ECPrivateKeyGetS);
```

Comment:
What will happen if I call `ExportParameters(includePrivate: true)` on object that was constructed with public key only?

Reply:
This question, to me, is "how do Java exceptions work from the JNI call?"

Reply:
Any Java method that can throw an exception needs to be check with `(*env)->ExceptionOccurred(env)`, see http://journals.ecs.soton.ac.uk/java/tutorial/native1.1/implementing/error.html